core/templates/pages/exploration-editor-page/modal-templates/lost-changes-modal.controller.spec.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for LostChangesModalController.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+import { LostChangeObjectFactory } from
+  'domain/exploration/LostChangeObjectFactory';
+import { UtilsService } from 'services/utils.service';
+
+describe('Lost Changes Modal Controller', () => {
+  let $scope = null;
+  let $uibModalInstance = null;
+  let $log = null;
+  let logSpy = null;
+  const explorationId = '0';
+  const lostChanges = [{
+    cmd: 'add_state',
+    state_name: 'State name',
+  }];
+
+  beforeEach(angular.mock.module('oppia', ($provide) => {
+    const ugs = new UpgradedServices();
```

Comment:
Hi @marianazangrossi, could you try not to add more of UpgradedService, we are getting rid of upgraded services. Instead, use TestBed and add those services to providers array and use TestBed.get(...) to retrieve instances of it. Please let me know if you want a more detailed explanation of how to use Angular services while testing ajs code.

Reply:
Hi @srijanreddy98, do you have any example of this new approach?

Reply:
Don't mind, I think I got it!

Reply:
Done!